src/ray/object_manager/object_manager.cc

```diff
@@ -1,156 +1,309 @@
 #include "object_manager.h"
 
+namespace asio = boost::asio;
+
 namespace ray {
 
-ObjectManager::ObjectManager(boost::asio::io_service &io_service,
+// TODO(hme): Clean up commented logs once multi-threading integration is completed.
+// Note: Current implementation has everything needed for concurrent transfers,
+// but concurrency is currently disabled for integration purposes. Concurrency is
+// disabled by running all asio components on the main thread (main_service).
+ObjectManager::ObjectManager(asio::io_service &main_service,
+                             std::unique_ptr<asio::io_service> object_manager_service,
                              ObjectManagerConfig config,
-                             std::shared_ptr<ray::GcsClient> gcs_client)
-    : object_directory_(new ObjectDirectory(gcs_client)), work_(io_service_) {
+                             std::shared_ptr<gcs::AsyncGcsClient> gcs_client)
+    // TODO(hme): Eliminate knowledge of GCS.
+    : client_id_(gcs_client->client_table().GetLocalClientId()),
+      object_directory_(new ObjectDirectory(gcs_client)),
+      object_manager_service_(std::move(object_manager_service)),
+      work_(*object_manager_service_),
```

Comment:
Why is this necessary? If this is only necessary for testing purposes, let's remove it from the source code.